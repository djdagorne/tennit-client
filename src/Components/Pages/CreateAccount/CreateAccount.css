
.form-section label {
    font-size:14px;
    display: block;
    margin:3vw;
}



input[type~=text],
input[type~=url],
input[type~=email],
input[type~=password],
select,
textarea{
}