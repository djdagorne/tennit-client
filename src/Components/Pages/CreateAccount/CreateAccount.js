import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import './CreateAccount.css';

class CreateAccount extends Component {
    constructor(props){
        super(props);
        this.state = {
            loggedIn: this.props.loggedIn,
            listingChecked: false,
        };
    };

    toggleListingSection(){
        this.setState({
            listingChecked: !this.state.listingChecked
        });
    }


    render(){
        return(
            <div className="popup sign-up">
                <div className="popup_inner">
                    <h3>Sign Up</h3>
                    <form id="sign-up">

                    <div className="form-section">
                        <label htmlFor="email">Your email</label>
                        <input 
                            type="email" 
                            name="email" 
                            placeholder="smithy@smithmail.com" 
                            /* required */
                        /> 
                    </div>

                    <div className="form-section">
                        <label htmlFor="password">Your password</label>
                        <input 
                            type="password" 
                            name="password" 
                            placeholder="*******" 
                            /* required */
                        />
                    </div>

                    <div className="form-section">
                        <label htmlFor="first-name">First Name</label>
                        <input 
                            type="text" 
                            name="first-name" 
                            placeholder="Smithy" 
                            /* required */
                        />
                    </div>

                    <div className="form-section">
                        <label htmlFor="last-name">Last Name</label>
                        <input 
                            type="text" 
                            name="last-name" 
                            placeholder="Smitherson" 
                            /* required */
                        />
                    </div>

                    <div className="form-section-dropdown">
                        <label htmlFor="gender">Gender</label>
                        <select name="gender" 
                        /* required */>
                            <option value="none">please pick one</option>
                            <option value="male">male</option>
                            <option value="female">female</option>
                            <option value="other">prefer not to say</option>
                        </select>
                    </div>

                    <div className="form-section-dropdown">
                        <label htmlFor="looking-for-gender">Looking for</label>
                        <select name="looking-for-gender" 
                        /* required */>
                            <option value="none">please pick one</option>
                            <option value="male">male</option>
                            <option value="female">female</option>
                            <option value="other">either</option>
                        </select>
                    </div>

                    <div className="form-section">
                        <label htmlFor="phone">Phone Number</label>
                        <input 
                            type="text" 
                            name="phone" 
                            placeholder="555-555-5555" 
                            /* required */
                        />
                    </div>

                    <div className="form-section">
                        <label htmlFor="age">Age</label>
                        <input 
                            type="text" 
                            name="age" 
                            placeholder="18" 
                            min="18" 
                            /* required */
                        />
                    </div>

                    <div className="search-segnment-dropdown">
                        <label htmlFor="country">country:</label>
                        <select name="country" /* required */>
                            <option value="none">please pick one</option>
                            <option>Canada</option>
                            <option>United States of America</option>
                        </select>
                    </div>

                    <div className="form-section">
                        <label htmlFor="provence">Provence or State</label>
                        <input 
                            type="text" 
                            name="provence" 
                            placeholder="Enter provence/state" 
                            /* required */
                        />
                    </div>

                    <div className="form-section">
                        <label htmlFor="city">City</label>
                        <input 
                            type="text" 
                            name="city" 
                            placeholder="Enter city name" 
                            /* required */
                        />
                    </div>

                    <div className="form-section">
                        <label htmlFor="image">Display your apartment image, link here</label>
                        <input 
                                type="url" 
                                name="image" 
                                placeholder="https://imagehost.com/image.jpg" 
                                /* required */
                            />
                    </div>

                    <div className="form-section">
                        <label htmlFor="user-details">Details</label>
                        <textarea 
                            rows="5" 
                            name="user-details" 
                            placeholder="Tell us about yourself! What is the first thing you want potential partners to know?" 
                            /* required */
                        />
                    </div>    

                    <div className="listing-section">
                        <label className="listing-section" htmlFor="listing-boolean">List your place?</label>
                        <input 
                            className="listing-section"
                            type="checkbox" 
                            name="listing-boolean" 
                            onClick={()=> this.toggleListingSection(this.state.listingChecked)}
                        />
                    </div>

                    {this.state.listingChecked ? 
                        <div className="listing-details">
                            <div className="form-section">
                                <label htmlFor="address">Address</label>
                                <input 
                                    type="text" 
                                    name="address" 
                                    placeholder="123 Street Road"
                                />
                            </div>

                            <div className="form-section">
                                <label htmlFor="neighborhood">Neighborhood (optional)</label>
                                <input 
                                    type="text" 
                                    name="neighborhood" 
                                    placeholder="eg financial district"
                                />
                            </div>

                            <div className="form-section">
                                <label htmlFor="rent">Monthly rent cost (per person)</label>
                                <input 
                                    type="text" 
                                    name="rent" 
                                    placeholder="750"
                                />
                            </div>

                            <div className="form-section">
                                <label htmlFor="home-details">Details</label>
                                <textarea 
                                    rows="5" 
                                    name="home-details" 
                                    placeholder="Got any ground rules? Pets? Feng Shui? Start the convo here!"
                                />
                            </div>  
                        </div> 
                        : null
                    }

                    <Link className="cancel button" to="/" onClick={this.props.closePopup}><button className="button">Cancel</button></Link>
                    <Link type="submit" to="/home" onClick={this.props.toggleLogIn}><button className="button">Submit</button></Link>

                </form>
                </div>
            </div>
        )
    }
}

export default CreateAccount;