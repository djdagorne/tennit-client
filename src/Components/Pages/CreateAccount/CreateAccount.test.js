import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CreateAccount from './CreateAccount';
import '../../../setupTest';



it('renders without crashing', () => {
    const wrapper = shallow(<CreateAccount />);
    expect(toJson(wrapper)).toMatchSnapshot();
})
