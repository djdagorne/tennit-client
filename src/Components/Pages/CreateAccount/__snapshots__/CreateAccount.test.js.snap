// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="popup"
>
  <div
    className="popup-inner"
  >
    <button
      className="close-popup"
      onClick={[Function]}
    >
      X
    </button>
    <h3
      className="banner-text header-two"
    >
      Sign Up
    </h3>
    <form
      id="create-account"
      onSubmit={[Function]}
    >
      <div
        className="form-item"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="password"
        >
          password
        </label>
        <input
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="firstname"
        >
          First Name
        </label>
        <input
          name="firstname"
          onChange={[Function]}
          placeholder="Smithy"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="lastname"
        >
          Last Name
        </label>
        <input
          name="lastname"
          onChange={[Function]}
          placeholder="Smitherson"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="age"
        >
          Age
        </label>
        <input
          name="age"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-item-dropdown"
      >
        <label
          htmlFor="usergender"
        >
          Gender
        </label>
        <select
          name="usergender"
          onChange={[Function]}
          required={true}
        >
          <option>
            please pick one
          </option>
          <option
            value="male"
          >
            male
          </option>
          <option
            value="female"
          >
            female
          </option>
          <option
            value="other"
          >
            other
          </option>
        </select>
      </div>
      <div
        className="form-item-dropdown"
      >
        <label
          htmlFor="prefgender"
        >
          Looking for
        </label>
        <select
          name="prefgender"
          onChange={[Function]}
          required={true}
        >
          <option>
            please pick one
          </option>
          <option
            value="male"
          >
            male
          </option>
          <option
            value="female"
          >
            female
          </option>
          <option
            value="other"
          >
            either
          </option>
        </select>
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="province"
        >
          Province
        </label>
        <input
          name="province"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="city"
        >
          City
        </label>
        <input
          name="city"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="image"
        >
          Apartment Image
        </label>
        <input
          name="image"
          onChange={[Function]}
          placeholder="as hyperlink"
          required={true}
          type="url"
        />
      </div>
      <div
        className="form-item"
      >
        <label
          htmlFor="userblurb"
        >
          Details
        </label>
        <textarea
          name="userblurb"
          onChange={[Function]}
          placeholder="Tell us about yourself! What is the first thing you want potential partners to know?"
          required={true}
          rows="5"
        />
      </div>
      <div
        className="checkbox-wrap"
      >
        <label
          className="listing-section"
          htmlFor="listing"
        >
          List your place?
        </label>
        <input
          className="listing-section"
          name="listing"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div />
      <div
        className="button-wrap"
      >
        <button
          className="rounded-button"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
